src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/action/index/TransportIndexRollupAction.kt

```diff
@@ -54,13 +57,19 @@ class TransportIndexRollupAction @Inject constructor(
     private val log = LogManager.getLogger(javaClass)
 
     override fun doExecute(task: Task, request: IndexRollupRequest, listener: ActionListener<IndexRollupResponse>) {
-        IndexRollupHandler(client, listener, request).start()
+        val userStr = client.threadPool().threadContext.getTransient<String>(ConfigConstants.OPENDISTRO_SECURITY_USER_INFO_THREAD_CONTEXT)
+        val user = resolveUser(User.parse(userStr))
+        val userName = client.threadPool().threadContext.getTransient<String>(ConfigConstants.INJECTED_USER)
```

Comment:
Is this pattern also used in Alerting?

Reply:
I don't think alerting use this.
Basically, this is used because we do index rollup in runner, and we should not update user object because of that call. It will update user name to plugin.

Reply:
Where is INJECTED_USER coming from?

Reply:
in rollup runner index rollup method
[code link](https://github.com/opendistro-for-elasticsearch/index-management/pull/394/files?file-filters%5B%5D=.kt#diff-9d27b0547d80839891dd1315fd5491e47db004d340770bcadfcdedd131ffedbbR390)

Reply:
I would prefer to define instead a variable like `shouldBackfillPermissions` or something along the line instead to make it clear the intent of doing this.

Reply:
Also I am not sure I get this entirely, why can't `putRollup` method just use the user object defined here `val user = resolveUser(User.parse(userStr))` to update the rollup object always? Shouldn't that solve the issue of backfill and also ensure the rollup stored object always have the permissions defined the last updated user?

Reply:
+1, using user is confusing

Reply:
changed to use a new element in thread context `INDEX_MANAGEMENT_PLUGIN_INTERNAL` to indicate if the request is from outside user or plugin runner.
if the value of this element is true, then we will not update user object.